chore(docs): Rework CI docs (#31988)
microsoft/playwright: docs/src/ci-intro.md

Playwright tests can be ran on any CI provider. In this section we will cover running tests on GitHub using GitHub actions. If you would like to see how to configure other CI providers check out our detailed doc on Continuous Integration.

#### You will learn

> * langs: python, java, csharp

Reviewer: just double check this makes sense for all langs except js or it should also have js?

Author: Yes this is intended. There's a separate "You will learn" list for JS that has different contents in https://github.com/microsoft/playwright/pull/31988/files#diff-49b43d48c93c3dc274299a0604e3a7c8ea1ed9891a230f3125e9992867d2ed3fL13-L14.

    - uses: actions/checkout@v4
    - uses: actions/setup-node@v4
      with:
        node-version: 18
        node-version: lts/*
    - name: Install dependencies
      run: npm ci
    - name: Install Playwright Browsers
      run: npx playwright install --with-deps
    - name: Run Playwright tests
      run: npx playwright test
    - uses: actions/upload-artifact@v4

>       if: ${{ !cancelled() }}

Reviewer 2: revert that?

Author: I've copied this over from the "on push/pull_request" recipe in `ci.md`. Let's try to use the same snippet between the two - which one do you prefer?

Reviewer 2: `if: ${{ !cancelled() }}` is what we changed it to based on https://github.com/microsoft/playwright/pull/30102

Author: hmm, then that missed one of the recipes. I'll update it!

Author: done in https://github.com/microsoft/playwright/pull/31988/commits/acbcbf1b472be396429dd8b49d1058eb2b235745

Author: I think I got confused here because I copied over from `create-playwright`. Looks like we missed that when updating, here's a PR to also use `!cancelled()` there: https://github.com/microsoft/create-playwright/pull/128

        name: playwright-report
        path: playwright-report/
        retention-days: 30
```

> If this doesn't make sense to you, give GitHub's [Understanding GitHub Actions](https://docs.github.com/en/actions/learn-github-actions/understanding-github-actions) guide a read. Looking at the list of steps in `jobs.test.steps`, you can see that the workflow performs these steps:

Reviewer: same comment as below regarding wording if this doesnt make sense

Author: done in https://github.com/microsoft/playwright/pull/31988/commits/34ccc7a1e17b7d114bd3018344ef7ea3864ce280

      with:
        distribution: 'temurin'
        java-version: '17'

>     - name: Ensure browsers are installed

Reviewer 3: Revert this? The step actually triggers build of the project (mvn compile) and then installs the artifacts into local maven cache. It does *not* install browsers, they will be downloaded on first playwright run.

      run: mvn -B install -D skipTests --no-transfer-progress

>     - name: Install Playwright

Reviewer 3: ```suggestion
    - name: Ensure browsers are installed
```

Author: good point, not sure what went wrong here. I've copied over the snippet from `ci.md` to keep them in sync. https://github.com/microsoft/playwright/pull/31988/commits/0be85a9ae65c679d35fe2e93f0b3576f758bd7b0

      run: dotnet test
```

> If this doesn't make sense to you, give GitHub's [Understanding GitHub Actions](https://docs.github.com/en/actions/learn-github-actions/understanding-github-actions) guide a read.

Reviewer: i dont like the wording here. it reads as if you are not sure if you explained it well. 
Best to put something like To learn more about GitHub Actions or something along those lines

Author: ```suggestion
To learn more about this, see ["Understanding GitHub Actions"](https://docs.github.com/en/actions/learn-github-actions/understanding-github-actions).
```

How do you like this?

Reviewer: Perfect. Now just to change it in intro doc too

      run: dotnet test
```

To learn more about this, see ["Understanding GitHub Actions"](https://docs.github.com/en/actions/learn-github-actions/understanding-github-actions).

Looking at the list of steps in `jobs.test.steps`, you can see that the workflow performs these steps:

1. Clone your repository

> 2. Install language dependencies

Reviewer 3: 3. Build project ?

Author: yes! done in https://github.com/microsoft/playwright/pull/31988/commits/59ae200e0cda728b51bc2a464b8fe7a14ef16ebe, and i've also updated the step names to be the same across the different CI snippets.
